Editorial updates to Overview.md (#144)
Wasm-DSL/spectec: spectec/doc/Overview.md

We also define *values* as a syntactic subset of instructions,
which are the ones that cannot be reduced any further:

> SpecTec automatically recognises that the type `val` defines a subtype of `instr` because `CONST valtype const` was already defined to be an instance of `instr`.

Reviewer: ```suggestion
SpecTec automatically recognises that the type `val` defines a subtype of `instr`
because it contains a subset of its cases.
```

hence it only matches values.

> `SELECT` is defined by two cases with different side conditions inspecting the selector value `c`.
> `=/=` is a built-in operator expressing inequality of natural numbers.

Reviewer: ```suggestion
`SELECT` is defined by two cases with different side conditions inspecting the selector value `c`;
where `=/=` expresses inequality.
```
